packages/permission-controller/src/PermissionController.ts

```diff
@@ -755,6 +764,18 @@ export class PermissionController<
       `${controllerName}:revokePermissions` as const,
       this.revokePermissions.bind(this),
     );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateCaveat` as const,
+      (origin, target, caveatType, caveatValue) => {
+        this.updateCaveat(
+          origin,
+          target,
+          caveatType as ExtractAllowedCaveatTypes<ControllerPermissionSpecification>,
```

Comment:
Hmm, why do we have to typecast here? Is there a way that we can avoid this?

Reply:
See https://github.com/MetaMask/core/pull/1071#discussion_r1082902728

Reply:
Er, right, I should have scrolled up 😅